src/app/projects/project-edit/project-edit.component.ts

```diff
@@ -23,41 +14,66 @@ import { Project } from "../models/project.model";
 })
 export class ProjectEditComponent implements OnInit {
 
+  @ViewChild('f') slForm: NgForm;
+
+  editMode;
   project;
   tempId;
-  constructor(private ar: ActivatedRoute,
-              private putProject: ProjectServiceService,
-              private router: Router) {
-                ar.params.subscribe( param => {
-                  this.tempId = param;
-                  });
+  errorMessage;
 
+  constructor(private route: ActivatedRoute,
+    private putProject: ProjectServiceService,
+    private authService: AuthService,
+    private router: Router) {
+    route.params.subscribe(param => {
+      this.tempId = param;
+    });
   }
 
-    editProject(event, projectName, linkImg, shortDesc, fullDesc, goals, result, involved, budget, type){
-      event.preventDefault();
-      let projectTemp: Project = new Project(
-        projectName.value, linkImg.value, shortDesc.value, 
-        fullDesc.value, goals.value, result.value, involved.value, type.value, budget.value);
+  actProject(form: NgForm) {
+    const value = form.value;
+    value.budget = value.budget || 0;
 
-      let mock = {name: projectName.value}
+    let projectTemp: Project = new Project(
+      this.authService.getNickname(),
+      value.projectName, value.image,
+      value.desc, value.goals, value.result, value.budget);
 
-      this.putProject.putProject(this.tempId.id, mock)
-      .subscribe(
+    let projectEdit: EditProject = new EditProject(
```

Comment:
Correct way to do this is:
1. create interface `EditProject` instead of class;
2. expect value to be instance of EditProject